File: app/src/main/java/spit/spdagger2/otherExample/di/module/ActivityModule.java
package spit.spdagger2.otherExample.di.module;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;
import spit.spdagger2.otherExample.di.ActivityContext;

/**
 * Created by janisharali on 08/12/16.
 */

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return mActivity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }
}

File: app/src/main/java/spit/spdagger2/main/kotlinexample/di/modules/ActivityModule.kt
package spit.spdagger2.main.kotlinexample.di.modules

import android.app.Activity
import dagger.Module
import dagger.Provides
import spit.spdagger2.main.kotlinexample.ui.main.MainContract
import spit.spdagger2.main.kotlinexample.ui.main.MainPresenter

@Module
class ActivityModule(private var activity: Activity) {

    @Provides
    fun provideActivity(): Activity {
        return activity
    }

    @Provides
    fun providePresenter(): MainContract.Presenter {
        return MainPresenter()
    }

}